persistent-postgresql/Database/Persist/Postgresql/Util.hs

```diff
@@ -0,0 +1,30 @@
+{-# LANGUAGE OverloadedStrings #-}
+
+module Database.Persist.Postgresql.Util (
+  mkPostgresUpdateText
+) where
+
+import Data.Monoid ((<>))
+import Data.Text (Text)
+import qualified Data.Text as T
+
+import Database.Persist (FieldNameDB, PersistEntity (..), PersistUpdate(..), Update(..), updateUpdate, fieldDB, FieldDef)
+
+
+mkPostgresUpdateText :: PersistEntity record => (FieldNameDB -> Text) -> (Text -> Text) -> Update record -> Text
+mkPostgresUpdateText escapeName refColumn x =
```

Comment:
So I'm not sure about this function! At least, not as a general purpose function. We only call it in the `mkBulkUpsertQuery`, which suggests to me that it should be local to that function. I'm also not sure that the name makes sense to me - this function is turning an `Update record` into the actual assignment statement, so something like `RecordField +=. 1` would become `"table_name"."field" = EXCLUDED."field" + 1`.

So, this is *only* valid during the `ON CONFLICT DO UPDATE` part of a query - and it does also mean that we're using the `EXCLUDED` for the source, rather than the pre-existing database table. We're calling this on the `[Update rec]`, which is *not* what we want! The `[Update rec]` is supposed to be a list of `Update`s that are *not* dependent on the inserted record.

See the comment on the parameter:

```haskell
  -- | A list of the updates to apply that aren't dependent on the record being inserted.
  [Update record] ->
```

I think we acn just use `Util.mkUpdateText`, like the MySQL version does.

Reply:
This is good feedback!  I think I must be doing something else wrong to work around this, then, because isn't a query like this:

```
INSERT INTO "item"("name", "description", "price", "quantity") 
VALUES (?, ?, ?, ?), (?, ?, ?, ?), (?, ?, ?, ?) 
ON CONFLICT ("name") 
DO UPDATE SET "quantity"=EXCLUDED."quantity"+? 
WHERE (EXCLUDED."description"=?)
```

the type of query we want?  When I run the query without referencing the `EXCLUDED` table, the SQL bugs out.

Reply:
Like am I just working around something i don't understand?

Reply:
Yeah. It's giving an error because the `"field_name"` is going to be ambiguous, and in a `ON CONFLICT DO UPDATE` there's going to *also* be the `EXCLUDED` table available with the same field names, so now we *have* to prefix the update with either the table name or the `EXCLUDED` name.

So I think we can do the same thing as we did with the `Prefix_` type - accept that value, and if it is `PrefixExcluded`, then we'll make it `EXCLUDED`, otherwise we prefix with the table name.